nengo/dists.py

```diff
@@ -148,20 +148,29 @@ class UniformHypersphere(Distribution):
 
     Parameters
     ----------
+    min_magnitude : Number
+        The closed lower bound of the uniform hypersphere distribution;
+        such that the sampled vector magnitudes >= min_magnitude.
+        The default value and minimum value for min_magnitude is 0.
+        The maximum value for min_magnitude is 1. Values outside this range
+        will be clipped to this range.
     surface : bool
         Whether sample points should be distributed uniformly
         over the surface of the hyperphere (True),
         or within the hypersphere (False).
+        Note: Setting surface to True generates sample points from the
+        hypersphere surface irregardless of what the value of min_magnitude is.
         Default: False
 
     """
 
-    def __init__(self, surface=False):
+    def __init__(self, min_magnitude=0, surface=False):
+        self.min_magnitude = min(max(min_magnitude, 0), 1)
         self.surface = surface
 
     def __repr__(self):
-        return "UniformHypersphere(%s)" % (
-            "surface=True" if self.surface else "")
+        return "UniformHypersphere(min_magnitude=%f%s)" % (
```

Comment:
Might want to use ```%g``` here instead of ```%f```.  Not sure if we've been consistent on that elsewhere, though.